### DISCUSSION TITLE
"docker pull paketobuildpacks/builder:full" takes very long time

### DISCUSSION THREAD

**Discussion by ivose** [paketo-buildpacks/feedback#178] [Builder Team]
Can somebody know why i must wait over an half of hour it?

**Comment by dmikusa** [comment]
It's a [big image](https://hub.docker.com/layers/paketobuildpacks/builder/full/images/sha256-47c63418204fdb030601093df5797f7ee8a0d7044ade0f2796eda3678bbb1aaf?context=explore), both layers (100) and size (750M compressed). That takes time to download.

Do you need to use the full image? The base builder image is about half that size.

At the end of the day, these images have a lot of build tools and libraries to build your code. These are comparable to other build environments in a container. It just takes time to download the bits.

Also, to clarify the large containers are the *build* images. There is a separate *run* image which will be smaller than the build image, because it doesn't have all the dev tools. The image that is produced by the build process for your application will be based on the run image so it can be much smaller. For example, typical images produced from the base run image around 80M + your app size. Tiny is about 20M + your app size. There's even a static image that's only a couple of MB + your app size.
